src/bank.rs

```diff
@@ -222,19 +234,25 @@ impl Bank {
     }
 
     fn update_signature_status_with_last_id(
-        &self,
+        last_ids_sigs: &mut HashMap<Hash, (SignatureStatusMap, u64)>,
         signature: &Signature,
         result: &Result<()>,
         last_id: &Hash,
     ) {
-        if let Some(entry) = self.last_ids_sigs.write().unwrap().get_mut(last_id) {
+        if let Some(entry) = last_ids_sigs.get_mut(last_id) {
             Self::update_signature_status(&mut entry.0, signature, result);
         }
     }
 
     fn update_transaction_statuses(&self, txs: &[Transaction], res: &[Result<()>]) {
+        let mut last_ids = self.last_ids_sigs.write().unwrap();
```

Comment:
Here you left it as a method on Bank. Inconsistent!

Reply:
@garious its not.  the first function is for 1 item.  this is for a bunch of transactions.